torch/jit/__init__.py

```diff
@@ -423,58 +423,65 @@ def trace(*args, **kwargs):
+                         support those types and shapes. example_inputs may also be a single
+                         Tensor in which case it is automatically wrapped in a tuple
 
     Keyword arguments:
         optimize (bool, optional): whether or not to apply optimizations.  Default: ``True``.
+        check_trace (bool, optional): check if the same inputs run through
+                                      traced code produce the same outputs. Default: ``True``. You might want
+                                      to disable this if, for example, your network contains non-
+                                      deterministic ops or if you are sure that the network is correct despite
+                                      a checker failure.
+
         check_inputs (list of tuples. optional): A list of tuples of input arguments that should be used
                                                  to check the trace against what is expected. Each tuple
                                                  is equivalent to a seet of input arguments that would
                                                  be specified in `args`. For best results, pass in a
                                                  set of checking inputs representative of the space of
                                                  shapes and types of inputs you expect the network to see.
                                                  If not specified, the original `args` is used for checking
-        disable_checks (bool, optional): Pass bool True to disable the aforementioned checking. You might
-                                         want to do this if, for example, your network contains non-
-                                         deterministic ops or if you are sure that the network is
-                                         correct despite a checker failure.
         check_tolerance (float, optional): Floating-point comparison tolerance to use in the checker procedure.
                                            This can be used to relax the checker strictness in the event that
                                            results diverge numerically for a known reason, such as operator fusion.
 
-        >>> @jit.trace(torch.rand(1))
-        ... def f(x):
-        ...     return x * 2
-    """
-    def wrapper(func):
-        if not _enabled:
-            return func
-        executor_options = {'optimize': True}
-        for name in executor_options:
-            executor_options[name] = kwargs.pop(name, executor_options[name])
-
-        check_inputs = kwargs.pop('check_inputs', None)
-        disable_checks = kwargs.pop('disable_checks', False)
-        check_tolerance = kwargs.pop('check_tolerance', 1e-5)
-
-        if len(kwargs) != 0:
-            raise TypeError("got unexpected keyword arguments: {}".format(", ".join(kwargs.keys())))
+    Returns:
+        A torch.jit.ScriptModule object with a single forward() method containing the traced code.
```

Comment:
I don't think we should be talking about `forward()` too much, given that this will be called on functions too. This might confuse people into thinking that they should do `fn.forward(...)` after they trace it.